Sources/Random.swift

```diff
@@ -6,66 +6,92 @@
 //  Copyright © 2016-2017 Károly Lőrentey.
 //
 
-import Foundation
-#if os(Linux) || os(FreeBSD)
-  import Glibc
-#endif
-
-
 extension BigUInt {
-    //MARK: Random Integers
-
-    /// Create a big integer consisting of `width` uniformly distributed random bits.
+    /// Create a big unsigned integer consisting of `width` uniformly distributed random bits.
     ///
-    /// - Returns: A big integer less than `1 << width`.
-    /// - Note: This function uses `arc4random_buf` to generate random bits.
-    public static func randomInteger(withMaximumWidth width: Int) -> BigUInt {
-        guard width > 0 else { return 0 }
-
-        let byteCount = (width + 7) / 8
-        assert(byteCount > 0)
-
-        let buffer = UnsafeMutablePointer<UInt8>.allocate(capacity: byteCount)
-      #if os(Linux) || os(FreeBSD)
-        let fd = open("/dev/urandom", O_RDONLY)
-        defer {
-            close(fd)
-        }
-        let _ = read(fd, buffer, MemoryLayout<UInt8>.size * byteCount)
-      #else
-        arc4random_buf(buffer, byteCount)
-      #endif
-        if width % 8 != 0 {
-            buffer[0] &= UInt8(1 << (width % 8) - 1)
+    /// - Parameter width: The maximum number of one bits in the result.
+    /// - Parameter generator: The source of randomness.
+    /// - Returns: A big unsigned integer less than `1 << width`.
+    public static func randomInteger<RNG: RandomNumberGenerator>(withMaximumWidth width: Int, using generator: inout RNG) -> BigUInt {
+        var result = BigUInt.zero
+        var bitsLeft = width
+        var i = 0
+        while bitsLeft >= Word.bitWidth {
+            result[i] = generator.next()
+            i += 1
+            bitsLeft -= Word.bitWidth
         }
-        defer {
-            buffer.deinitialize(count: byteCount)
-            buffer.deallocate()
+        if bitsLeft > 0 {
+            let mask: Word = (1 << bitsLeft) - 1
+            result[i] = (generator.next() as Word) & mask
```

Comment:
This could possibly fail on 32-bit systems where `Word` would be a UInt32. Since `generator.next()` will always be a `UInt64` the conversion will fail.
`Word(truncatingIfNeeded: BinaryInteger)` could be used instead.

Reply:
If `Word` is `UInt32`, this will call [the generic `next`](https://developer.apple.com/documentation/swift/randomnumbergenerator/2996533-next), which is defined for all types conforming to `FixedWidthInteger & UnsignedInteger`.

Reply:
I must have overlooked that in the documentation.